multigen/pipes.py

```diff
@@ -188,6 +195,53 @@ def gen(self, inputs):
         return image
 
 
+class MaskedIm2ImPipe(Im2ImPipe):
+    _class = MaskedStableDiffusionImg2ImgPipeline
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._mask = None
+        self._image_painted = None
+        self._original_image = None
+        self._mask_blur = None
+
+    def setup(self, original_image=None, image_painted=None, mask=None, blur=4, **kwargs):
+        self._original_image = original_image
+        self._image_painted = image_painted
+        # there are two options:
+        # 1. mask is provided
+        # 2. mask is computed from difference between original_image and image_painted
+        if image_painted is not None:
+            neq = numpy.any(numpy.array(self._original_image) != numpy.array(self._image_painted), axis=-1)
+            mask = neq.astype(numpy.uint8) * 255
+        else:
+            assert mask is not None
+        self._mask = mask
+        self._image_painted = image_painted
+        input_image = image_painted if image_painted is not None else original_image
+        input_image = numpy.array(input_image)
+        super().setup(fimage=None, image=input_image / input_image.max(), **kwargs)
+        pil_mask = mask
+        if not isinstance(self._mask, PIL.Image.Image):
+            pil_mask = PIL.Image.fromarray(mask)
+            if pil_mask.mode != "L":
+                pil_mask = pil_mask.convert("L")
+        mask_blur = pil_mask.filter(ImageFilter.GaussianBlur(radius=blur))
+        mask_blur = numpy.array(mask_blur)
+        self._mask_blur = numpy.tile(mask_blur / mask_blur.max(), (3, 1, 1)).transpose(1,2,0)
+
+    def gen(self, inputs):
+        inputs = inputs.copy()
+        inputs.update(mask=self._mask)
+        img_gen = super().gen(inputs)
+
+        # compose with original using mask
+        img_compose = self._mask_blur * img_gen + (1 - self._mask_blur) * self._original_image
+        # convert to PIL image
+        img_compose = PIL.Image.fromarray(img_compose.astype(numpy.uint8))
```

Comment:
`Image` from `PIL` is imported. No need to use `PIL.Image` here and to import `PIL` as a whole above